pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -343,39 +399,308 @@ func (r *ReconcileVirtualMachineImport) addWatchForImportPod(instance *v2vv1.Vir
 	)
 }
 
-func (r *ReconcileVirtualMachineImport) importDisks(provider provider.Provider, instance *v2vv1.VirtualMachineImport, mapper provider.Mapper, vmName types.NamespacedName) error {
+// convertGuest starts a Job to run virt-v2v on the target VM
+func (r *ReconcileVirtualMachineImport) convertGuest(provider provider.Provider, instance *v2vv1.VirtualMachineImport, vmName types.NamespacedName) (bool, error) {
+	// find the vmspec
+	vmSpec := &kubevirtv1.VirtualMachine{}
+	err := r.client.Get(context.TODO(), vmName, vmSpec)
+	if err != nil {
+		return false, err
+	}
+
+	configMap, err := r.findLibvirtDomainConfigMap(vmName)
+	if err != nil {
+		return false, err
+	}
+	if configMap == nil {
+		configMap, err = r.makeLibvirtDomainConfigMap(instance, vmSpec)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	job, err := r.findGuestConversionJob(vmName)
+	if err != nil {
+		return false, err
+	}
+	// the job doesn't exist, so create it
+	if job == nil {
+		job = r.makeGuestConversionJobSpec(vmSpec)
+		// Set VirtualMachineImport instance as the owner and controller so it'll be cleaned up
+		// when the instance is removed.
+		if err := controllerutil.SetControllerReference(instance, job, r.scheme); err != nil {
+			return false, err
+		}
+		err := r.client.Create(context.TODO(), job)
+		if err != nil {
+			return false, err
+		}
+
+		processingCond := conditions.NewProcessingCondition(string(v2vv1.ConvertingGuest), "Running virt-v2v", corev1.ConditionTrue)
+		err = r.upsertStatusConditions(types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, processingCond)
+		if err != nil {
+			return false, err
+		}
+
+		// Update progress to converting guest:
+		if err = r.updateProgress(instance, progressConvertingGuest); err != nil {
+			return false, err
+		}
+	}
+
+	if job.Status.Active > 0 {
+		return false, nil
+	}
+
+	if job.Status.Failed > 0 {
+		err := r.endGuestConversionAsFailed(provider, instance, "virt-v2v job failed")
+		if err != nil {
+			return false, err
+		}
+		return false, nil
+	}
+
+	return job.Status.Succeeded > 0, nil
+}
+
+// findGuestConversionJob finds the guest conversion job for a given VM, returning nil if it can't be found.
+func (r *ReconcileVirtualMachineImport) findGuestConversionJob(vmName types.NamespacedName) (*batchv1.Job, error) {
+	jobList := &batchv1.JobList{}
+	matchingLabels := client.MatchingLabels(map[string]string{VMLabel: vmName.Name})
+	err := r.client.List(context.TODO(), jobList, matchingLabels)
```

Comment:
namespace isn't needed in this case?

Reply:
Replacing this entirely with a Job manager in the same vein as the other resource managers.